src/main/java/net/neoforged/neoforge/registries/NeoForgeRegistries.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.registries;
+
+import com.mojang.serialization.Codec;
+import java.util.function.Supplier;
+import net.minecraft.core.Registry;
+import net.minecraft.network.syncher.EntityDataSerializer;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.ItemDisplayContext;
+import net.neoforged.neoforge.common.advancements.critereon.ICustomItemPredicate;
+import net.neoforged.neoforge.common.conditions.ICondition;
+import net.neoforged.neoforge.common.crafting.IngredientType;
+import net.neoforged.neoforge.common.loot.IGlobalLootModifier;
+import net.neoforged.neoforge.common.world.BiomeModifier;
+import net.neoforged.neoforge.common.world.StructureModifier;
+import net.neoforged.neoforge.fluids.FluidType;
+import net.neoforged.neoforge.internal.versions.neoforge.NeoForgeVersion;
+import net.neoforged.neoforge.registries.holdersets.HolderSetType;
+
+/**
+ * A class that exposes static references to all vanilla and Forge registries.
+ * Created to have a central place to access the registries directly if modders need.
+ * It is still advised that if you are registering things to use {@link RegisterEvent} or {@link DeferredRegister}, but queries and iterations can use this.
+ */
+public class NeoForgeRegistries {
+    // Custom forge registries
+    static final DeferredRegister<EntityDataSerializer<?>> DEFERRED_ENTITY_DATA_SERIALIZERS = DeferredRegister.create(Keys.ENTITY_DATA_SERIALIZERS, Keys.ENTITY_DATA_SERIALIZERS.location().getNamespace());
+    public static final Registry<EntityDataSerializer<?>> ENTITY_DATA_SERIALIZERS = DEFERRED_ENTITY_DATA_SERIALIZERS.makeRegistry(registryBuilder -> registryBuilder.sync(true));
+    static final DeferredRegister<Codec<? extends IGlobalLootModifier>> DEFERRED_GLOBAL_LOOT_MODIFIER_SERIALIZERS = DeferredRegister.create(Keys.GLOBAL_LOOT_MODIFIER_SERIALIZERS, Keys.GLOBAL_LOOT_MODIFIER_SERIALIZERS.location().getNamespace());
+    public static final Registry<Codec<? extends IGlobalLootModifier>> GLOBAL_LOOT_MODIFIER_SERIALIZERS = DEFERRED_GLOBAL_LOOT_MODIFIER_SERIALIZERS.makeRegistry(registryBuilder -> {});
```

Comment:
Since registries are immediately resolved, I think we should just remove registry creation from DR's.  A registry should be created prior to the creation of a DR for that registry, not vice versa.

Reply:
the DR stuff for custom registries is just a helper so you don't have to do everything manually

Reply:
Depends what you mean by doing stuff manually... Surely the registry has to be created anyway? Could just be a static method call somewhere given that everything is happening statically now...

Reply:
Yeah, should just be a static on RegistryBuilder, not stapled to DR

Reply:
Registration of registries will remain in the `NewRegistryEvent` because we want to avoid registration triggered by classloading. DR is just a helper, but you can also register to `NewRegisterEvent` directly (as I have made NeoForge do in https://github.com/Matyrobbrt/NeoForge/commit/3011de89b90c777118ead8e0752b49582e9cff42) - the choice is yours.